vector/src/main/java/im/vector/app/features/home/HomeActivityViewModel.kt

```diff
@@ -130,6 +133,47 @@ class HomeActivityViewModel @AssistedInject constructor(
                 .launchIn(viewModelScope)
     }
 
+    /**
+     * Handle threads migration. The migration includes:
+     * - Notify users that had io.element.thread enabled from labs
+     * - Re-Enable m.thread to those users (that they had enabled labs threads)
+     * - Handle migration when threads are enabled by default
+     */
+    private fun initThreadsMigration() {
```

Comment:
how would you feel about extracting this logic to its own `ThreadsMigrator` class?  I'm wondering if in the future we would want to add more migrations to this `ViewModel`

Reply:
Sounds good !

Reply:
I am not sure if it would be more clear to extract the logic while it mainly invokes  `_viewEvents.post ` events, and I am afraid it might be confusing

Reply:
I was imagining split between the view event creation and the logic of deciding the migration status
not a blocker for me as it's not something we do often in the current codebase 

```kotlin
sealed interface Result { 
  object Migrated: Result
  data class Migrate(migrateExistingSession: Boolean)
}

class ThreadsMigator {
  fun migrate() : Result { }
}


// viewmodel
fun initThreadsMigration() {
   val event = when(val result = threadsMigrator.migrate()) {
     Migrated -> NotifyUserForThreadsMigration
     ...
   }
   post(event)
}

```

has the benefit of writing tests simpler for both migration and viewmodel ^^^ but does introduce extra boilerplate for bridging the layers